cadquery/occ_impl/shapes.py

```diff
@@ -1379,6 +1397,20 @@ def _geomAdaptorH(self) -> Tuple[BRepAdaptor_Curve, BRepAdaptor_HCurve]:
 
         return curve, BRepAdaptor_HCurve(curve)
 
+    def close(self) -> Union["Edge", "Wire"]:
+        """
+        Close an Edge
+        """
+        rv: Union[Wire, Edge]
+
+        if not self.IsClosed():
+            e = Edge.makeLine(self.endPoint(), self.startPoint())
+            rv = Wire.assembleEdges((self, e))
+        else:
+            rv = self
+
+        return rv
+
```

Comment:
Would this go better on Mixin1D so it could close Wires as well?

Reply:
I don't think that this part is generic - I think it is nice to keep closed edges as `Edge`. I could add a Wire specific close to the Wire class.

Reply:
> I don't think that this part is generic - I think it is nice to keep closed edges as Edge.

If self is an Edge, and `self.IsClosed()` is True, it will just return self, right? Edges will stay Edges?

> I could add a Wire specific close to the Wire class.

I have no need for it, just thought I saw an opportunity to generalise this with no extra code.

Reply:
Again, what's the point of using a Mixing if you are dispatching in it on the specific type of self.

Reply:
Ah, sorry, I tested moving this to `Mixin1D.close` locally, but I must have stuffed up my mypy check, it was all green. I see the problem now.